### ISSUE TITLE
Improved progress bar

### ISSUE THREAD

**Issue by gjcooper** [gjcooper/samplers#48]
**Is your feature request related to a problem? Please describe.**
When messages/warnings are displayed it "interrupts" the progress bar which makes the display messy

**Describe the solution you'd like**
Something that overwrites the current output with the messages and progress.

**Describe alternatives you've considered**
Leave as is and improve warning messages only

**Additional context**
See possible implementation test
```r
for (i in 1:5) {
  message('\033[F\r', paste('Running', i))
  message('\r', strrep("=",i), appendLF = FALSE)
  Sys.sleep(1)
}
```

**Comment by gjcooper** [comment]
Rather than overwrite the whole screen as suggested it may be better to "hold" the warnings/messages until after the current stage has finished (or at least all but the first one perhaps)

**Comment by gjcooper** [comment]
Closed by 6703f00
